pytket/qir/conversion/conversion.py

```diff
@@ -493,50 +493,98 @@ def circuit_to_module(
-                condition_name = command.args[0].reg_name
-
-                def condition_block_true() -> None:
-                    """
-                    Populate recursively the module with the contents of the conditional
-                    sub-circuit when the condition is True.
-                    """
-                    if op.value == 1:
-                        self.circuit_to_module(conditional_circuit, module)
+                if op.width == 1:  # only one conditional bit
+                    conditional_circuit = self._rebase_op_to_gateset(
+                        op.op, command.args[op.width :]
+                    )
+                    condition_bit_index = command.args[0].index[0]
+                    condition_name = command.args[0].reg_name
+
+                    def condition_block_true() -> None:
+                        """
+                        Populate recursively the module with the contents of the
+                        conditional sub-circuit when the condition is True.
+                        """
+                        if op.value == 1:
+                            self.circuit_to_module(conditional_circuit, module)
+
+                    def condition_block_false() -> None:
+                        """
+                        Populate recursively the module with the contents of the
+                        conditional sub-circuit when the condition is False.
+                        """
+                        if op.value == 0:
+                            self.circuit_to_module(conditional_circuit, module)
+
+                    if condition_name in self.ssa_vars:
+
+                        ssabool = module.builder.call(
+                            self.read_bit_from_reg,
+                            [
+                                self.ssa_vars[condition_name],
+                                pyqir.const(self.qir_int_type, condition_bit_index),
+                            ],
+                        )
 
-                def condition_block_false() -> None:
-                    """
-                    Populate recursively the module with the contents of the conditional
-                    sub-circuit when the condition is False.
-                    """
-                    if op.value == 0:
-                        self.circuit_to_module(conditional_circuit, module)
+                        module.module.builder.if_(
+                            ssabool,
+                            true=lambda: condition_block_true(),  # type: ignore
+                            false=lambda: condition_block_false(),  # type: ignore
+                        )
 
-                if condition_name in self.ssa_vars:
+                    else:
+                        ValueError(
+                            """circuit contruction with this condition
+                            not yet supported"""
+                        )
+                        # this should be an assertion when working
```

Comment:
Not sure what this comment means.

Reply:
See @cqc-melf
[clean up conditional handling](https://github.com/CQCL/pytket-qir/pull/37/commits/43eefb9ba2e33b72efc8026785a99f7596aaf385)
[43eefb9](https://github.com/CQCL/pytket-qir/pull/37/commits/43eefb9ba2e33b72efc8026785a99f7596aaf385)